File: workspace/java/src/test/java/com/dslplatform/ocd/values/TestOneUUIDValue.java
package com.dslplatform.ocd.values;

import com.dslplatform.client.Bootstrap;
import com.dslplatform.ocd.test.TypeTester;
import com.dslplatform.ocd.values.OneUUIDInValue.OneUUIDValue;
import com.dslplatform.patterns.ServiceLocator;
import java.util.*;
import org.junit.*;
import static org.junit.Assert.*;

public class TestOneUUIDValue {
    private static ServiceLocator locator;

    @BeforeClass
    public static void setUpClass() throws Exception {
        locator = Bootstrap.init(TestOneUUIDValue.class.getResourceAsStream("dsl-project.ini"));
    }

    @AfterClass
    public static void tearDownClass() throws Exception {
        locator.resolve(java.util.concurrent.ExecutorService.class).shutdown();
        locator = null;
    }

    @Before
    public void setUp() throws Exception {
    }

    @After
    public void tearDown() throws Exception {
    }

    @Test
    public void testFieldType() throws NoSuchFieldException {
        assertTrue(TypeTester.testField(OneUUIDValue.class, "oneUUID")
                .resultEquals(UUID.class));
    }

    @Test
    public void testGetterType() throws NoSuchMethodException {
        assertTrue(TypeTester.testGetter(OneUUIDValue.class, "getOneUUID")
                .resultEquals(UUID.class));
    }

    @Test
    public void testSetterType() throws NoSuchMethodException {
        assertTrue(TypeTester.testSetter(OneUUIDValue.class, "setOneUUID", UUID.class)
                .resultEquals(OneUUIDValue.class));
    }
}

File: workspace/scala/src/test/scala/com/dslplatform/ocd/values/TestOneUUIDValue.scala
package com.dslplatform.ocd.values

import OneUUIDInValue.OneUUIDValue
import com.dslplatform.api.patterns.ServiceLocator
import org.scalatest._
import scala.reflect.runtime.universe._

class TestOneUUIDValue extends SpecExtensions {
  def `Test property field type: java.util.UUID`() =
    checkType(
      typeOf[java.util.UUID]
    , typeOf[OneUUIDValue].member("oneUUID": TermName).asMethod.returnType
    )

  def `Test property default value: java.util.UUID.randomUUID`() {
    assert(OneUUIDValue().oneUUID.getClass == classOf[java.util.UUID]);
  }
}
